### PR TITLE
Fix Deposit Feed test

### PR DESCRIPTION
**Pull request by maurelian**
**Description**
1. Adds a return value check in the Dummy contract used for testing Deposits made by a contract.
2. This led me to identify a test that was passing incorrectly.
3. And I added two more test cases that were lacking.

### CODE REVIEW

File: packages/contracts/test/L1/DepositFeed.spec.ts [hunk 1]
```diff
@@ -164,18 +209,55 @@ describe('DepositFeed', () => {
         })
       })
 
+      it('when a contract deposits a transaction with an ETH value.', async () => {
```
Comment (protolambda) [inline packages/contracts/test/L1/DepositFeed.spec.ts:212]: The dummy contract forwards ETH, as opposed to spending existing balance. During execution it's effectively the same (contract balance increases and decreases), but do we need to test this the other way?

File: packages/contracts/test/L1/DepositFeed.spec.ts [hunk 2]
```diff
@@ -164,18 +209,55 @@ describe('DepositFeed', () => {
...
+        // Deploy a dummy contract so we can impersonate it
+        const dummy = await (await ethers.getContractFactory('Dummy')).deploy()
+        await dummy.deployed()
+        // this is not emitting an event!
```
Comment (protolambda) [inline packages/contracts/test/L1/DepositFeed.spec.ts:216]: It does decode an awaited event below however, is this comment wrong?

**Review by protolambda — COMMENTED** [review]
Wondering about one comment, and if there's a better way to test the contract paying the deposited ETH. Looks good otherwise :+1:

File: packages/contracts/test/L1/DepositFeed.spec.ts (hunk 2 above)
Comment (maurelian) [reply packages/contracts/test/L1/DepositFeed.spec.ts:216]: my bad, this was a debugging comment that got left in.

Comment (protolambda) [reply packages/contracts/test/L1/DepositFeed.spec.ts]: Removed the old comment in 55bf74ba0b9a48d536b98bdfff8c4d23d8a10b1d

**Review by protolambda — APPROVED** [review]
LGTM!

File: packages/contracts/test/L1/DepositFeed.spec.ts (hunk 1 above)
Comment (maurelian) [reply packages/contracts/test/L1/DepositFeed.spec.ts:212]: I think it's fine. There's zero difference in the execution context of the DepositFeed right?

**Pull request merged**